File: remote/src/main/java/com/kelsos/mbrc/interactors/PlayerInteractorImpl.java
package com.kelsos.mbrc.interactors;

import com.google.inject.Inject;
import com.kelsos.mbrc.annotations.PlayerAction;
import com.kelsos.mbrc.dto.BaseResponse;
import com.kelsos.mbrc.services.api.PlayerService;

import rx.Observable;

public class PlayerInteractorImpl implements PlayerInteractor {
  @Inject private PlayerService api;
  @Override
  public Observable<BaseResponse> execute(@PlayerAction.Action String action) {
    return api.performPlayerAction(action);
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/interactors/PlayerInteractorImpl.kt
package com.kelsos.mbrc.interactors

import com.kelsos.mbrc.annotations.PlayerAction.Action
import com.kelsos.mbrc.dto.BaseResponse
import com.kelsos.mbrc.services.api.PlayerService
import rx.Observable
import javax.inject.Inject

class PlayerInteractorImpl
@Inject constructor(private val api: PlayerService) : PlayerInteractor {
  override fun performAction(@Action action: String): Observable<BaseResponse> {
    return api.performPlayerAction(action)
  }
}
